card/mod/01_history/lib/card/action.rb

```diff
@@ -112,25 +119,35 @@ def field_index field
       end
     end
 
-    def new_value_for field
-      ch = card_changes.find_by(field: field_index(field))
-      ch && ch.value
+    def interpret_field field
+      case field
+      when :content then :db_content
+      when :cardtype then :type_id
+      else field.to_sym
+      end
     end
 
-    def change_for field
-      card_changes.where 'card_changes.field = ?', field_index(field)
+    def interpret_value field, value
+      case field.to_sym
+      when :type_id
+        value && value.to_i
+      when :cardtype
+        type_card = value && Card.quick_fetch(value.to_i)
+        type_card && type_card.name.capitalize
+      else value
+      end
     end
 
     def new_type?
-      new_value_for :type_id
+      value :type_id
     end
 
     def new_content?
-      new_value_for :db_content
+      value :db_content
     end
 
     def new_name?
-      new_value_for :name
+      value :name
     end
```

Comment:
I've been working out this "new" vocab.  I understand where it came from, but it's an easier translation to the database if we refer to the "new" value as just the "value" of the change.  (it's the value field of the change row, after all")